pkg/odo/util/cmdutils.go

```diff
@@ -95,26 +91,32 @@ func PrintComponentInfo(client *occlient.Client, currentComponentName string, co
 		}
 
 		// Cut off the last newline and output
-		output = output[:len(output)-1]
-		log.Describef("Environment Variables:\n", output)
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			log.Describef("Environment Variables:\n", output)
+		}
+
 	}
 
 	// Storage
 	if len(componentDesc.Spec.Storage) > 0 {
 
 		// Retrieve the storage list
-		storages, err := localConfig.StorageList()
+		storages, err := storage.List(client, currentComponentName, applicationName)
```

Comment:
@kadel @girishramnani @mik-dass So as per the discussions, should i remove this change, and only keep bounds check here? and the `state` part will be implemented as a part of feature request?